File: app/src/main/java/cn/ifafu/ifafu/mvp/syllabus_setting/SyllabusSettingContract.java
package cn.ifafu.ifafu.mvp.syllabus_setting;

import cn.ifafu.ifafu.data.entity.SyllabusSetting;
import cn.ifafu.ifafu.mvp.base.i.IModel;
import cn.ifafu.ifafu.mvp.base.i.IPresenter;
import cn.ifafu.ifafu.mvp.base.i.IView;

public class SyllabusSettingContract {

    interface View extends IView {

        void initRecycleView(SyllabusSetting setting);

    }

    public interface Presenter extends IPresenter {

    }

    interface Model extends IModel {
        SyllabusSetting getSetting();
    }

}

File: app/src/main/java/cn/ifafu/ifafu/mvp/syllabus_setting/SyllabusSettingContract.kt
package cn.ifafu.ifafu.mvp.syllabus_setting

import android.widget.ImageView
import cn.ifafu.ifafu.base.mvp.IModel
import cn.ifafu.ifafu.base.mvp.IPresenter
import cn.ifafu.ifafu.base.mvp.IView
import cn.ifafu.ifafu.entity.SyllabusSetting

class SyllabusSettingContract {

    interface View : IView {

        fun initRecycleView(items: List<Any>)

        fun showPicturePicker()

        fun showColorPicker(setting: SyllabusSetting, ivColor: ImageView)
    }

    interface Presenter : IPresenter {

        fun onPictureSelect(uri: String)

        suspend fun onFinish()
    }

    interface Model : IModel {

        fun getSetting(): SyllabusSetting

        suspend fun save(setting: SyllabusSetting)
    }

}
